src/lib/DevicesProvider.js

```diff
@@ -3,64 +3,85 @@ import React, { Component } from 'react'
 import { findById } from 'lib/utils'
 import P from 'prop-types'
 import * as R from 'ramda'
-import { guid } from 'utils'
+import { objectToUrl } from 'utils'
+import { read, update, destroy, create } from 'lib/xhr-helpers'
 
 export const DevicesContext = React.createContext()
 
 const toggleActive = R.over(R.lensProp('active'), R.not)
 
+const readDevices = urlParams =>
+  read(`http://localhost/api/devices?${objectToUrl(urlParams)}`)
+
+const updateDevice = async (id, item) => {
+  const response = await update(`http://localhost/api/devices/edit/${id}`, item)
+  return response
+}
+
+const deleteDevice = id => destroy(`http://localhost/api/devices/remove/${id}`)
+
 export class DeviceProvider extends Component {
   constructor(props) {
     super(props)
-    const cachedDevices = localStorage.getItem('devices')
-
     this.state = {
-      devices: JSON.parse(cachedDevices) || [],
+      devices: [],
       onSave: this.onSaveHandler,
       onDelete: this.deleteHandler,
       onChangeActive: this.onChangeActive,
       onSubmit: this.onDeviceFormSubmit,
     }
   }
 
-  onDeviceFormSubmit = item => {
-    const { devices } = this.state
+  async componentDidMount() {
+    const dev = await readDevices()
+    this.setState({ devices: dev.data })
+  }
 
-    const newDevices = devices.concat(R.assoc('id', guid(), item))
+  onDeviceFormSubmit = async item => {
+    const { devices } = this.state
+    const response = await create('http://localhost/api/devices/create', item)
```

Comment:
Please follow the same pattern as the rest of the fetch calls. Create a new function outside the class and call this instead. Also don't name your variables "response". This is too generic. In this case you know what `create` call will return